lib/adapters/xhr.js

```diff
@@ -27,6 +27,10 @@ module.exports = function xhrAdapter(config) {
       }
     }
 
+    if (utils.isFormData(requestData) && utils.isStandardBrowserEnv()) {
+      delete requestHeaders['Content-Type']; // Let the browser set it
```

Comment:
Shouldn't this be done [further down](https://github.com/axios/axios/blob/6ee101b4b5074cf5f0d7f4d007ec181827635f01/lib/adapters/xhr.js#L158) in order to use normalised headers?

Using this here would only catch the specific casing `Content-Type` and not `content-type`

Reply:
Probably yes, but this PR only brings back the [removed code](https://github.com/axios/axios/pull/3785/files#) and adds an additional environment check to fix the regression bug. Axios headers helper is far away from perfection, and it should be deeply refactored to be really careless, not just locally.
Unfortunately, the PR #4215 where the headers helper is reworked is still not merged, so it would be better to keep it as it was for a while.
https://github.com/axios/axios/pull/4215/files#diff-e3e3dd5daf3f63345305a7397f39f6adce58f3ff9b987a96da808b060079958b